searx/engines/google_news.py

```diff
@@ -16,74 +45,155 @@
 }
 
-# search-url
+# engine dependent config
+
 categories = ['news']
-paging = True
+paging = False
 language_support = True
-safesearch = True
+use_locale_domain = True
 time_range_support = True
-number_of_results = 10
+safesearch = True # not really, but it is not generated by google
 
-search_url = 'https://www.google.com/search'\
-    '?{query}'\
-    '&tbm=nws'\
-    '&gws_rd=cr'\
-    '&{search_options}'
-time_range_attr = "qdr:{range}"
-time_range_dict = {'day': 'd',
-                   'week': 'w',
-                   'month': 'm',
-                   'year': 'y'}
-
-
-# do search-request
 def request(query, params):
-
-    search_options = {
-        'start': (params['pageno'] - 1) * number_of_results
-    }
-
-    if params['time_range'] in time_range_dict:
-        search_options['tbs'] = time_range_attr.format(range=time_range_dict[params['time_range']])
-
-    if safesearch and params['safesearch']:
-        search_options['safe'] = 'on'
-
-    params['url'] = search_url.format(query=urlencode({'q': query}),
-                                      search_options=urlencode(search_options))
-
-    if params['language'] != 'all':
-        language = match_language(params['language'], supported_languages, language_aliases).split('-')[0]
-        if language:
-            params['url'] += '&hl=' + language
+    """Google-News search request"""
+
+    language, country, lang_country = get_lang_country(
+        # pylint: disable=undefined-variable
+        params, supported_languages, language_aliases
+    )
+    subdomain = 'news.google.com'
+
+    if params['time_range']: # in time_range_dict:
+        query += ' ' + time_range_dict[params['time_range']]
+
+    query_url = 'https://'+ subdomain + '/search' + "?" + urlencode({
+        'q': query,
+        'hl': language,
+        'lr': "lang_" + language,
+        'ie': "utf8",
+        'oe': "utf8",
+        'ceid' : "%s:%s" % (country, language),
+        'gl' : country,
+    })
+
+    if params['safesearch']:
+        query_url += '&' + urlencode({'safe': filter_mapping[params['safesearch']]})
+
+    params['url'] = query_url
+    logger.debug("query_url --> %s", query_url)
+
+    # en-US,en;q=0.8,en;q=0.5
+    params['headers']['Accept-Language'] = (
+        lang_country + ',' + language + ';q=0.8,' + language + ';q=0.5'
+        )
+    logger.debug("HTTP header Accept-Language --> %s",
+                 params['headers']['Accept-Language'])
+    params['headers']['Accept'] = (
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        )
+    #params['google_subdomain'] = subdomain
 
     return params
 
 
-# get response from search-request
 def response(resp):
+    """Get response from google's search request"""
     results = []
 
+    # detect google sorry
+    resp_url = urlparse(resp.url)
+    if resp_url.netloc == 'sorry.google.com' or resp_url.path == '/sorry/IndexRedirect':
+        raise RuntimeWarning('sorry.google.com')
```

Comment:
See
https://github.com/searx/searx/blob/f310305c54c3cd1d9fc74f09453294edbd2b5486/searx/engines/google.py#L205-L211

It should be [`SearxEngineCaptchaException`](https://github.com/searx/searx/blob/f310305c54c3cd1d9fc74f09453294edbd2b5486/searx/exceptions.py#L76-L83) instead of `RuntimeWarning` : Searx will then stop sending request for long time.

The default suspend time is 1 day, but the `suspended_time` parameter can provide another value (in second).